server/src/main/java/org/opensearch/index/engine/ReadOnlyEngine.java

```diff
@@ -223,7 +253,17 @@ protected final OpenSearchDirectoryReader wrapReader(
 
     protected DirectoryReader open(IndexCommit commit) throws IOException {
         assert Transports.assertNotTransportThread("opening index commit of a read-only engine");
-        return new SoftDeletesDirectoryReaderWrapper(DirectoryReader.open(commit), Lucene.SOFT_DELETES_FIELD);
+        DirectoryReader reader;
+        if (isExtendedCompatibility()) {
+            reader = DirectoryReader.open(commit, this.minimumSupportedVersion.luceneVersion.major, null);
+        } else {
+            reader = DirectoryReader.open(commit);
+        }
+        return new SoftDeletesDirectoryReaderWrapper(reader, Lucene.SOFT_DELETES_FIELD);
+    }
+
+    private boolean isExtendedCompatibility() {
+        return this.minimumSupportedVersion.before(Version.CURRENT.minimumIndexCompatibilityVersion());
```

Comment:
`getExtendedCompatibilitySnapshotVersion` will return null if the FF is not enabled, which is what is invoked when the new ctor is called from IndexService.  Would this not throw a NPE when FF is disabled?

Reply:
Good spot. The current code manages to avoid this case, but doesn't explicitly protect against it. I'll fix this and add a related test case as well

Reply:
Implicitly protected against an NPE by adding defaulting behavior in `IndexSettings` when it computes `extendedCompatibilitySnapshotVersion`